eval/helm/main.py

```diff
@@ -0,0 +1,181 @@
+    )
+    fabric.launch()
+
+    checkpoint_dir = Path(checkpoint_dir)
+    check_valid_checkpoint_dir(checkpoint_dir)
+
+    with open(checkpoint_dir / "lit_config.json") as fp:
+        config = Config(**json.load(fp))
+
+    checkpoint_path = checkpoint_dir / "lit_model.pth"
+    logger.info(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
+    with fabric.init_module(empty_init=True), quantization(quantize):
+        model = GPT(config)
+
+    with lazy_load(checkpoint_path) as checkpoint:
+        model.load_state_dict(checkpoint, strict=quantize is None)
+
+    model.eval()
+    model = fabric.setup(model)
+
+    tokenizer = Tokenizer(checkpoint_dir)
+
+    @app.post("/process")
+    async def process_request(input_data: ProcessRequest) -> ProcessResponse:
+        if input_data.seed is not None:
+            L.seed_everything(input_data.seed)
+        logger.info("Using device: {}".format(fabric.device))
+        encoded = tokenizer.encode(
+            input_data.prompt, bos=True, eos=False, device=fabric.device
+        )
+        prompt_length = encoded.size(0)
+        max_returned_tokens = prompt_length + input_data.max_new_tokens
+        assert max_returned_tokens <= model.config.block_size, (
+            max_returned_tokens,
+            model.config.block_size,
+        )  # maximum rope cache length
+
+        t0 = time.perf_counter()
+        tokens, logprobs, top_logprobs = generate(
+            model,
+            encoded,
+            max_returned_tokens,
+            max_seq_length=max_returned_tokens,
+            temperature=input_data.temperature,
+            top_k=input_data.top_k,
+        )
+
+        t = time.perf_counter() - t0
+
+        model.reset_cache()
+        output = tokenizer.decode(tokens)
+        tokens_generated = tokens.size(0) - prompt_length
+        logger.info(
+            f"Time for inference: {t:.02f} sec total, {tokens_generated / t:.02f} tokens/sec"
+        )
+
+        logger.info(f"Memory used: {torch.cuda.max_memory_reserved() / 1e9:.02f} GB")
+        generated_tokens = []
+        for t, lp, tlp in zip(tokens, logprobs, top_logprobs):
+            idx, val = tlp
+            tok_str = tokenizer.processor.decode([idx])
+            token_tlp = {tok_str: val}
+            generated_tokens.append(
+                Token(text=tokenizer.decode(t), logprob=lp, top_logprob=token_tlp)
+            )
+        logprobs_sum = sum(logprobs)
+        # Process the input data here
+        return ProcessResponse(
+            text=output, tokens=generated_tokens, logprob=logprobs_sum, request_time=t
+        )
+
+    @app.post("/tokenize")
+    async def tokenize(input_data: TokenizeRequest) -> TokenizeResponse:
+        logger.info("Using device: {}".format(fabric.device))
+        t0 = time.perf_counter()
+        encoded = tokenizer.encode(
+            input_data.text, bos=True, eos=False, device=fabric.device
```

Comment:
Are we sure about `bos=True` and `eos=False` for all models supported by lit-gpt?
Looks to me like these settings should be model specific unless I'm missing something (e.g. LLaMA 1 doesn't have a `bos` token)